cubano-httpeasy/src/main/java/org/concordion/cubano/driver/http/HttpEasyDefaults.java

```diff
@@ -1,96 +1,86 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
-import javax.net.ssl.HostnameVerifier;
-import javax.net.ssl.HttpsURLConnection;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.SSLSession;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
+import com.github.markusbernhardt.proxy.ProxySearch;
 
 /**
  * Allows setting of default properties used by all subsequent HttpEasy requests.
  *
  * @author Andrew Sumner
  */
 public class HttpEasyDefaults {
-    // Static values are set by RestRequestDefaults and apply to all requests
+    private static String baseUrl = "";
+    private static boolean trustAllCertificates = false;
+    private static List<String> sensitiveParameters = new ArrayList<>();
+
+    // Request authorisation
+    private static String authUser = null;
+    private static String authPassword = null;
+
+    // Proxy
+    private static ProxyConfiguration proxyConfiguration = ProxyConfiguration.MANUAL;
+    private static volatile ProxySelector proxySearch = null;
     private static Proxy proxy = Proxy.NO_PROXY;
     private static String proxyUser = null;
     private static String proxyPassword = null;
-    private static boolean bypassProxyForLocalAddresses = true;
-    private static String baseURI = "";
-    private static LogWriter defaultLogWriter = null;
+
+    // Logging
+    private static LogWriter defaultLogWriter = new LoggerLogWriter();
+    private static boolean logRequest = true;
+    private static boolean logRequestDetails = false;
+
 
     /**
-     * Create all-trusting certificate verifier.
-     *
+     * Create all-trusting certificate and host name verifier per HTTPS request.
+     * 
+     * @param trustAllCertificates Set to true to trust all certificates, the default is false
      * @return A self reference
      */
-    public HttpEasyDefaults trustAllCertificates() {
-        // Create a trust manager that does not validate certificate chains
-        TrustManager[] trustAllCerts = new TrustManager[]{
-            new X509TrustManager() {
-                public java.security.cert.X509Certificate[] getAcceptedIssuers() {
-                    return null;
-                }
-
-                public void checkClientTrusted(X509Certificate[] certs, String authType) {
-                }
-
-                public void checkServerTrusted(X509Certificate[] certs, String authType) {
-                }
-            }
-        };
-
-        // Install the all-trusting trust manager
-        try {
-            SSLContext sc = SSLContext.getInstance("SSL");
-            sc.init(null, trustAllCerts, new java.security.SecureRandom());
-            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
-        } catch (Exception e) {
-            HttpEasy.LOGGER.error(e.getMessage());
-        }
+    public HttpEasyDefaults trustAllCertificates(boolean trustAllCertificates) {
```

Comment:
Can we rename to `trustAllCertificatesForNextRequest` to be explicit?

Reply:
There are a number of items that can be set either as global default or per request.  If we're going to do it to one we should do it to all.  

My latest commit makes it a little more obvious as HttpEasy now uses Optional<?> for those fields that can override global defaults to improve readibility of the code.  This won't help end user so we can discuss this if you still think we need to change method names.